src/Sylius/Component/Promotion/Action/PromotionApplicator.php

```diff
@@ -38,11 +38,15 @@ public function __construct(ServiceRegistryInterface $registry)
      */
     public function apply(PromotionSubjectInterface $subject, PromotionInterface $promotion)
     {
+        $applyPromotion = false;
         foreach ($promotion->getActions() as $action) {
-            $this->getActionCommandByType($action->getType())->execute($subject, $action->getConfiguration(), $promotion);
+            $result = $this->getActionCommandByType($action->getType())->execute($subject, $action->getConfiguration(), $promotion);
+            $applyPromotion = $result ? true : false;
```

Comment:
Should be `$applyPromotion |= $result;`

We're missing one example:
1. First action was applied
2. Second was not applied
3. The promotion should be applied